src/braket/pennylane_plugin/braket_device.py

```
@@ -675,6 +676,107 @@ def _run_snapshots(self, snapshot_circuits, n_qubits, mapped_wires):
+                        f"and {freq_max*1e-9}, but recieved {freq}"
+                    )
+
+    def _validate_pulse_parameters(self, ev):
+        """Validates pulse input (ParametrizedEvolution) before converting to a PulseGate"""
+
+        # note: the pulse upload on the other side immediately checks that the max amplitude
+        # is not exceeded, so that check has not been included here
+
+        # confirm frequencies are constant and within the permitted frequency range for the channel
+        self._check_pulse_frequency_validity(ev)
+
+        # confirm all phase values are constant (or the qml.pulse.constant function)
+        callable_phase = [
+            pulse.phase
+            for pulse in ev.H.pulses
+            if (callable(pulse.phase) and pulse.phase != qml.pulse.constant)
+        ]
+
+        if callable_phase:
+            raise RuntimeError(
+                "Expected all phases to be constants or qml.pulse.constant, "
+                "but recieved callable(s)"
+            )
+
+        # ensure each ParametrizedEvolution/PulseGate contains at most one waveform per frame/wire
+        wires_used = []
+        for pulse in ev.H.pulses:
+            for wire in pulse.wires:
+                if wire in wires_used:
+                    raise RuntimeError(
+                        f"Multiple waveforms assigned to wire {wire} in the same "
+                        f"ParametrizedEvolution gate"
+                    )
+                wires_used.append(wire)
+
+        if ev.H.settings:
+            warnings.warn(
+                "The ParametrizedEvolution contains settings from an interaction term "
+                "`qml.pulse.transmon_interaction`. Please note that the settings passed to the "
+                "interaction term are not used for hardware upload. All parameters used in the "
+                "interaction term are set by the physical device."
+            )
```

I'm curious what people think about having this raise an exception rather than a warning. I.e.,
```python
H = qml.pulse.transmon_interaction(**settings, coupling=0.02)  # interaction term
H += qml.pulse.transmon_drive(0.18, 0., freq, wires=[0])  # drive term
op = qml.evolve(H)([], t=10.)
```
This would be a warning if run on hardware.
```python
H = qml.pulse.transmon_drive(0.18, 0., freq, wires=[0])  # drive term only
op = qml.evolve(H)([], t=10.)
```
This would not.

This has the advantage of being clear that the users interaction term cannot be used (I worry about users ignoring warnings). The drawback is that users will need to manually comment out the interaction term between simulator and hardware runs.

---

I personally think it's better to have a warning, as including an interaction term or not does not impact the results in any way, and I feel like it's not major enough that the program should exit. Happy to discuss though

---

I see your point about there being potential for confusion to users when setting interaction terms and ignoring warnings. Raising an exception is still quite severe, I'd probably be annoyed by this behavior as a user (one who knows that the interaction term has no effect though). Same for raising a warning, actually, but it seems like fair middle ground and sufficient to me.

---

@trbromley This is the only remaining question for this PR. Let me know what you think :) I'll merge it afterwards.

---

> the interaction term has no effect

I'm confused, doesn't it have an effect on simulator but not on hardware? I guess that's the potentially breaking behaviour I'm suggesting we avoid.

However, I don't want to be a blocker on this and we can always see if users find it confusing.

---

I see what you mean. I think I'll just merge this for now. We can add it as an open question in the PR we open to AWS.